zerver/views/realm_export.py

```diff
@@ -41,19 +57,20 @@ def export_realm(request: HttpRequest, user: UserProfile) -> HttpResponse:
     if limit_check >= EXPORT_LIMIT:
         raise JsonableError(_("Exceeded rate limit."))
 
-    # The RealmCount analytics table lets us efficiently get an
-    # estimate for the number of public stream messages in an
-    # organization. It won't match the actual number of messages in
-    # the export, because this measures the number of messages that
-    # went to a public stream at the time they were sent. Thus,
-    # messages that were deleted or moved between streams will be
+    # The RealmCount analytics table lets us efficiently get an estimate
+    # for the number of messages in an organization. It won't match the
+    # actual number of messages in the export, because this measures the
+    # number of messages that went to DMs / Group DMs / public or private
+    # channels at the time they were sent.
+    # Thus, messages that were deleted or moved between channels and
+    # private messages for which the users didn't consent for export will be
     # treated differently for this check vs. in the export code.
-    exportable_messages_estimate = sum(
-        realm_count.value
-        for realm_count in RealmCount.objects.filter(
-            realm=realm, property="messages_sent:message_type:day", subgroup="public_stream"
-        )
+    realm_count_query = RealmCount.objects.filter(
+        realm=realm, property="messages_sent:message_type:day"
     )
+    if export_type == RealmExport.EXPORT_PUBLIC:
+        realm_count_query.filter(subgroup="public_stream")
+    exportable_messages_estimate = sum(realm_count.value for realm_count in realm_count_query)
```

Comment:
In the case of "full export with user consent", this will consider all the messages sent, irrespective of the users who consented or not. Is it fine -- do maxm users consent in such export cases?

Reply:
Yeah, that's fine -- this is just an estimate.